middleware/loadshed.go

```diff
@@ -133,64 +119,28 @@ func (lq *latencyQueue) reSize() {
 
 	size := len(lq.sl)
 	if size > 5_000 {
-		// Each `latency` struct is 16bytes. So we can afford to have 5_000(80KB)
+		// Each `latency` struct is 8bytes. So we can afford to have 5_000(40KB)
 		half := size / 2
 		lq.sl = lq.sl[half:] // retain the latest half.
 	}
 }
 
-// todo: refactor this and its dependents.
-// currently they consume 9.04MB and 80ms as measured by the `BenchmarkAllMiddlewares` benchmark.
```

Comment:
This now does not appear in mem-profile.
The whole of `LoadShedder` middleware has `3MB` mem usage and its all in `wrappedHandler(w, r)`